scripts/pythondistdeps.py

```diff
@@ -334,83 +366,71 @@ def normalize_name(name):
                     spec = ('==', dist.py_version)
                     if spec not in py_deps[name]:
                         py_deps[name].append(spec)
-                deps = dist.requires()
+                deps = dist.requirements
                 if args.recommends:
-                    depsextras = dist.requires(extras=dist.extras)
-                    if not args.requires:
-                        for dep in reversed(depsextras):
-                            if dep in deps:
-                                depsextras.remove(dep)
-                    deps = depsextras
+                    deps = dist.extra_requirements
                 # console_scripts/gui_scripts entry points need pkg_resources from setuptools
-                if ((dist.get_entry_map('console_scripts') or
-                    dist.get_entry_map('gui_scripts')) and
+                if (dist.entry_points and
                     (lower.endswith('.egg') or
                      lower.endswith('.egg-info'))):
-                    # stick them first so any more specific requirement overrides it
-                    deps.insert(0, Requirement.parse('setuptools'))
+                    groups = set([ep.group for ep in dist.entry_points])
+                    if set(["console_scripts", "gui_scripts"]) & groups:
+                        # stick them first so any more specific requirement
+                        # overrides it
+                        deps.insert(0, Requirement('setuptools'))
                 # add requires/recommends based on egg/dist metadata
                 for dep in deps:
                     if normalized_names_require_pep503:
-                        dep_normalized_name = normalize_name(dep.project_name)
+                        dep_normalized_name = normalize_name(dep.name)
                     else:
-                        dep_normalized_name = dep.key
+                        dep_normalized_name = dep.name.lower().replace('_', '-')
 
                     if args.legacy:
-                        name = 'pythonegg({})({})'.format(pyver_major, dep.key)
+                        name = 'pythonegg({})({})'.format(pyver_major, dep.name.lower())
```

Comment:
Because dep.key is from setuptools' pkg_resources.Requirement and packaging.Requirement does not include it -- I'm not against creating a subclass of packing.Requirement here and including a key property that returns self.name.lower()